documentation/migration-guides/Az.12.0.0.0-migration-guide.md

```diff
@@ -0,0 +1,2769 @@
+# Migration Guide for Az 12.0.0
+
+## Az.Accounts
+
+### `Clear-AzConfig`
+
+- Cmdlet breaking-change will happen to all parameter sets
+  - Parameter `DisableErrorRecordsPersistence` will be deprecated, a new parameter `EnableErrorRecordsPersistence` will be added instead. Writing error records to file system will become opt-in instead of opt-out. This change will happen around May 2024
+  - This change is expected to take effect from Az.Accounts version: 2.X and Az version: 12.0.0
+
+#### Before
+```powershell
+Clear-AzConfig -DisableErrorRecordsPersistence
+```
+#### After
+```powershell
+error record will be disabled by default. New parameter 'EnableErrorRecordsPersistence' added to enable erro record.
+```
+
+
+### `Get-AzConfig`
+
+- Cmdlet breaking-change will happen to all parameter sets
+  - Parameter `DisableErrorRecordsPersistence` will be deprecated, a new parameter `EnableErrorRecordsPersistence` will be added instead. Writing error records to file system will become opt-in instead of opt-out. This change will happen around May 2024
+  - This change is expected to take effect from Az.Accounts version: 2.X and Az version: 12.0.0
+
+#### Before
+```powershell
+Get-AzConfig -DisableErrorRecordsPersistence 
+```
+#### After
+```powershell
+error record will be disabled by default. New parameter 'EnableErrorRecordsPersistence' added to enable erro record.
+```
+
+
+### `Update-AzConfig`
+
+- Cmdlet breaking-change will happen to all parameter sets
+  - Parameter `DisableErrorRecordsPersistence` will be deprecated, a new parameter `EnableErrorRecordsPersistence` will be added instead. Writing error records to file system will become opt-in instead of opt-out. This change will happen around May 2024
+  - This change is expected to take effect from Az.Accounts version: 2.X and Az version: 12.0.0
+
+#### Before
+```powershell
+Update-AzConfig -DisableErrorRecordsPersistence 
+```
+#### After
+```powershell
+error record will be disabled by default. New parameter 'EnableErrorRecordsPersistence' added to enable erro record.
+```
+
+
+## Az.Compute
+
+### `Get-AzVmss`
+
+- Cmdlet breaking-change will happen to all parameter sets
+  - Starting in May 2024 the "Get-AzVmss" cmdlet will no longer allow an empty value for resource group name and virtual machine scale set name.
+  - This change is expected to take effect from Az.Compute version: 8.0.0 and Az version: 12.0.0
+
+#### Before
+```powershell
+Get-AzVmss -ResourceGroupName ""
+# returned empty list
+```
+#### After
+```powershell
+Get-AzVmss -ResourceGroupName ""
+# will return error for empty string validation in the parameter 
+```
+
+
+### `New-AzGalleryImageDefinition`
+
+- Cmdlet breaking-change will happen to all parameter sets
+  - Starting in May 2024 the "New-AzGalleryImage" cmdlet will deploy with the Trusted Launch configuration by default. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch
```

Comment:
```suggestion
  - Starting in May 2024 the "New-AzGalleryImage" cmdlet will deploy with the Trusted Launch configuration and Gen2 Hyper V Generation by default. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch
```

Reply:
```suggestion
  - Starting in May 2024 the "New-AzGalleryImage" cmdlet will deploy with the Trusted Launch configuration and Gen2 Hyper V Generation by default. To know more about Trusted Launch, please visit [https://learn.microsoft.com/azure/virtual-machines/trusted-launch](/azure/virtual-machines/trusted-launch)
```

Reply:
This URL causes a hard-coded-locale and docs-link-absolute build warning in the docs repo.